app/views/users/_main_feed.html.erb

```diff
@@ -13,19 +13,15 @@
 <% end %>
 <% if @stories.present? %>
   <%= render "articles/single_story", story: @stories.first, featured: false %>
-  <% if @comments.present? %>
-    <% if user_signed_in? %>
-      <%= render "users/comments_section" %>
-    <% else %>
-      <%= render "users/comments_locked_cta" %>
-    <% end %>
-  <% end %>
-  <%= render partial: "articles/single_story", collection: @stories[1, @stories.size], as: :story, locals: { featured: false } %>
-  <div class="placeholder-div"></div>
-<% elsif @comments.present? %>
-  <% if user_signed_in? %>
+  <% if user_signed_in? && @comments.present? %>
```

Comment:
I spent some time testing this and _I think_ the new logic changes the feature behaviour a bit.

Previously when there were **comments present**, then we checked whether the user was signed in and showed the comment section or if they were not signed in then we showed the locked CTA.

But also, previously when there were **no comments present** we would just not display that section (neither the comments section box nor the CTA) to the user. 

However, with the change above if there are **no comments present** we continue showing the CTA box in its place. 

Please clarify that my observation of the changed behaviour is correct, because perhaps I misunderstood it. If the behaviour has changed I'm not sure if this is negligible for the product feature or whether its something we need to consider, but just wanted to bring it up for consideration.

Reply:
Once again, fantastic catch! I believe f8d4882 now has this logic restored (and bullet remains happy).

I'm a little troubled that this user-comment-section logic doesn't seem to be captured by any of our tests, I'll see if I can add some.